### PR TITLE
Not change file source status while changing cache path.

### PR DESCRIPTION
**Pull request by Chaoba**
` fileSource.deactivate()` will pause `MainResourceLoader` and so that it will not work for loading local resources.

This PR removes the process of changing fileSource status to keep `MainResourceLoader` running.

### CODE REVIEW

**Comment by tobrun** [comment]
@LukasPaczos as original implementer of this API you have a bit more background on the need to activate/deactivate the FileSource. Could you add some context?

**Comment by LukasPaczos** [comment]
We need to manually activate the file source here, at least we had to in the past. The native `setResourceCachePath` switches the calling thread to the `FileSourceThread` and resets the path and the database on that thread. If it would have been paused, the callback would never return.

That's why we need to activate the file source (and unpause the thread) to complete the operation and later do the cleanup and deactivate to keep the activation counter honest.

There's the [ChangeResourcesCachePathActivity](https://github.com/mapbox/mapbox-gl-native-android/blob/master/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/activity/offline/ChangeResourcesCachePathActivity.kt) example that should reproduce the issue on this branch and never be able to change the path.

**Comment by Chaoba** [comment]
> We need to manually activate the file source here, at least we had to in the past. The native `setResourceCachePath` switches the calling thread to the `FileSourceThread` and resets the path and the database on that thread. If it would have been paused, the callback would never return.

Do a test by deactivating filesource before changing cache path and it could work.

What confuses me is that when a FileSouce instance is created, the `resourceLoader` inside is in activated state while `FileSouce.isActivated()` return false.

If we activate the FileSouce before changing cache path and then deactivate it, `resourceLoader` will be in paused state, that's the reason why it will not load resource after changing cache path.

**Comment by LukasPaczos** [comment]
> the resourceLoader inside is in activated state while FileSouce.isActivated() return false

Maybe the activation counter in the `file_source.cpp` is incorrectly initialized after some recent changes?

**Comment by Chaoba** [comment]
@LukasPaczos Yeah, it is. Init the counter in the constructor method and the default state for `FileSource` is active now.

File: MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/storage/FileSource.java [hunk 1]
```diff
@@ -293,12 +293,7 @@ public static void setResourcesCachePath(@NonNull final String path,
     final Context applicationContext = Mapbox.getApplicationContext();
     final FileSource fileSource = FileSource.getInstance(applicationContext);
 
-    if (fileSource.isActivated()) {
```
Comment (LukasPaczos) [inline MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/storage/FileSource.java:296]: What happens now if we would try to change the cache path while we're processing tiles? What happens if the map is rendering and we call this method, are we going to crash or will we be able to safely reset the database pointer?

Comment (Chaoba) [reply MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/storage/FileSource.java:296]: Not seeing anything wrong during my test. @alexshalamov for visibility.

**Pull request merged**